app/models/observation.rb

```diff
@@ -1,13 +1,13 @@
 class Observation
-  attr_accessor :hash
+  attr_accessor :raw
   attr_accessor :hour
   attr_accessor :temperature
 
-  def self.new_from_hash(observationHash)
+  def self.new_from_raw(raw_observation)
     new.tap do |o|
-      o.hash = observationHash
-      o.hour = observationHash['date']['hour'].to_i
-      o.temperature = observationHash['tempi'].to_f
+      o.raw = raw_observation
+      o.hour = Time.zone.parse(raw_observation['date']['iso8601']).hour
+      o.temperature = raw_observation['temperature']
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.